### ISSUE TITLE
Only Point A is fixed in the path

### ISSUE THREAD

**Issue by zetta** [robertsambuena/delivery#4]
I've noticed in your solution that you have fixed point A and n-1, however this is not what we are looking for, only first point given as the input has to be fixed as a starting point, 2nd, 3rd, etc will be determined only if that is the closest path to take in order to visit all the drop-off locations.

**Comment by zetta** [comment]
So for example. 

this input 
```
[
  ["22.372080", "114.107876"],
  ["22.284418", "114.159509"],
  ["22.326441", "114.167810"],
  ["22.284418", "114.159509"]
]
```

should calculate the same path as if we send this. 
```
[
  ["22.372080", "114.107876"],
  ["22.284418", "114.159509"],
  ["22.284418", "114.159509"],
  ["22.326441", "114.167810"]
]
```

**Comment by robertsambuena** [comment]
Right, I'll work on the adjustments now. Thank you!

**Comment by robertsambuena** [comment]
Hello @zetta , I have a question regarding your sample inputs, based on googlemaps directions API, origin **`22.326441,114.167810`** is much closer to the starting point(**`22.372080,114.107876`**) than **`22.284418,114.159509`**. Does that mean the order should be

````
[
  ["22.372080", "114.107876"],
  ["22.326441", "114.167810"],
  ["22.284418", "114.159509"],
  ["22.284418", "114.159509"]
]
````

Correct me if I'm wrong as it will help on the logic of the implementation of the api. Thank you!

**Comment by robertsambuena** [comment]
I've pushed a new commit(**bc935b**) with the implementation I mentioned on the previous comment. Let me know what you think! @zetta

**Comment by zetta** [comment]
Sorry for not answering before, if one point is closer to the starting
point it doesn't mean that should be the next step, you have to measure the
length of the entire journey (path) to be able to decide if that's the
right path, you can't calculate the right path going step by step (small
steps).

Am I being clear?

**Comment by zetta** [comment]
In other words, getting shortest path step by step wont guarantee to get the shortest total distance

**Comment by robertsambuena** [comment]
Right, that make sense, I guess this would need a kind of travelling salesman solution correct?

**Comment by robertsambuena** [comment]
Hello @zetta , just want to inform you that I might need a little bit more time, probably on the weekends, to finish the task, this is because I might need to change my implementation drastically (TSP + distanceMatrix) and also because of my current work schedules. I'll be grinding this on the weekends and hopefully be done by then.

**Comment by robertsambuena** [comment]
Hello again @zetta , I did just pushed a new commit for the new implementation. Please have a look and let me know what you think. I hope the update will satisfy the requirements this time. Thank you!

**Comment by robertsambuena** [comment]
@zetta just a note, I updated  the `GOOGLE_MAPS_API_KEY` as I did reached the max limit on the other one, value is in here (https://github.com/robertsambuena/delivery/commit/4b235f40839c2dfed32853c07fdbb150136a4d20). Thanks!
